src/pages/workspace/reimburse/WorkspaceReimburseView.tsx

```diff
@@ -1,83 +1,56 @@
-            }),
-        ),
-        outputCurrency: PropTypes.string,
-        lastModified: PropTypes.number,
-    }).isRequired,
-
-    /** From Onyx */
+type WorkspaceReimburseViewOnyxProps = {
     /** Bank account attached to free plan */
-    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountPropTypes,
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
-    ...withLocalizePropTypes,
+    reimbursementAccount: OnyxEntry<OnyxTypes.ReimbursementAccount>;
 };
 
-const defaultProps = {
-    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountDefaultProps,
+type WorkspaceReimburseViewProps = WorkspaceReimburseViewOnyxProps & {
+    /** Policy values needed in the component */
+    policy: OnyxEntry<OnyxTypes.Policy>;
 };
 
-function WorkspaceReimburseView(props) {
+function WorkspaceReimburseView({policy, reimbursementAccount}: WorkspaceReimburseViewProps) {
     const styles = useThemeStyles();
-    const [currentRatePerUnit, setCurrentRatePerUnit] = useState('');
-    const viewAllReceiptsUrl = `expenses?policyIDList=${props.policy.id}&billableReimbursable=reimbursable&submitterEmail=%2B%2B`;
-    const distanceCustomUnit = _.find(lodashGet(props.policy, 'customUnits', {}), (unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
-    const distanceCustomRate = _.find(lodashGet(distanceCustomUnit, 'rates', {}), (rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
-    const {translate, toLocaleDigit} = props;
+    const [currentRatePerUnit, setCurrentRatePerUnit] = useState<string | undefined>('');
+    const viewAllReceiptsUrl = `expenses?policyIDList=${policy?.id ?? ''}&billableReimbursable=reimbursable&submitterEmail=%2B%2B`;
+    const distanceCustomUnit = Object.values(policy?.customUnits ?? {}).find((unit) => unit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+    const distanceCustomRate = Object.values(distanceCustomUnit?.rates ?? {}).find((rate) => rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
+    const {translate, toLocaleDigit} = useLocalize();
+    const {isOffline} = useNetwork();
 
-    const getUnitLabel = useCallback((value) => translate(`common.${value}`), [translate]);
+    const getUnitLabel = useCallback((value: Unit) => translate(`common.${value}`), [translate]);
 
     const getCurrentRatePerUnitLabel = useCallback(() => {
-        const customUnitRate = _.find(lodashGet(distanceCustomUnit, 'rates', '{}'), (rate) => rate && rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
-        const currentUnit = getUnitLabel(lodashGet(distanceCustomUnit, 'attributes.unit', CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES));
+        const customUnitRate = Object.values(distanceCustomUnit?.rates ?? {}).find((rate) => rate && rate.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
+        const currentUnit = getUnitLabel(distanceCustomUnit?.attributes.unit ?? CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES);
+        if (!customUnitRate) {
+            return undefined;
+        }
```

Comment:
Let's update `getUnitRateValue` typing instead, it looks like it already can handle the case where `customUnitRate` is undefined
```suggestion
```